x/tokenfactory/types/msgs.go

```diff
@@ -0,0 +1,204 @@
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid sender address (%s)", err)
+	}
+
+	if !m.Amount.IsValid() || m.Amount.Amount.Equal(sdk.ZeroInt()) {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, m.Amount.String())
+	}
+
+	return nil
+}
+
+func (m MsgMint) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&m))
+}
+
+func (m MsgMint) GetSigners() []sdk.AccAddress {
+	sender, _ := sdk.AccAddressFromBech32(m.Sender)
+	return []sdk.AccAddress{sender}
+}
+
+var _ sdk.Msg = &MsgBurn{}
+
+// NewMsgBurn creates a message to burn tokens
+func NewMsgBurn(sender string, amount sdk.Coin) *MsgBurn {
+	return &MsgBurn{
+		Sender: sender,
+		Amount: amount,
+	}
+}
+
+func (m MsgBurn) Route() string { return RouterKey }
+func (m MsgBurn) Type() string  { return TypeMsgBurn }
+func (m MsgBurn) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(m.Sender)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid sender address (%s)", err)
+	}
+
+	if !m.Amount.IsValid() || m.Amount.Amount.Equal(sdk.ZeroInt()) {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, m.Amount.String())
+	}
+
+	return nil
+}
+
+func (m MsgBurn) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&m))
+}
+
+func (m MsgBurn) GetSigners() []sdk.AccAddress {
+	sender, _ := sdk.AccAddressFromBech32(m.Sender)
+	return []sdk.AccAddress{sender}
+}
+
+// var _ sdk.Msg = &MsgForceTransfer{}
+
+// // NewMsgForceTransfer creates a transfer funds from one account to another
+// func NewMsgForceTransfer(sender string, amount sdk.Coin, fromAddr, toAddr string) *MsgForceTransfer {
+// 	return &MsgForceTransfer{
+// 		Sender:              sender,
+// 		Amount:              amount,
+// 		TransferFromAddress: fromAddr,
+// 		TransferToAddress:   toAddr,
+// 	}
+// }
+
+// func (m MsgForceTransfer) Route() string { return RouterKey }
+// func (m MsgForceTransfer) Type() string  { return TypeMsgForceTransfer }
+// func (m MsgForceTransfer) ValidateBasic() error {
+// 	_, err := sdk.AccAddressFromBech32(m.Sender)
+// 	if err != nil {
+// 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid sender address (%s)", err)
+// 	}
+
+// 	_, err = sdk.AccAddressFromBech32(m.TransferFromAddress)
+// 	if err != nil {
+// 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid address (%s)", err)
+// 	}
+// 	_, err = sdk.AccAddressFromBech32(m.TransferToAddress)
+// 	if err != nil {
+// 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid address (%s)", err)
+// 	}
+
+// 	if !m.Amount.IsValid() {
+// 		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, m.Amount.String())
+// 	}
+
+// 	return nil
+// }
+
+// func (m MsgForceTransfer) GetSignBytes() []byte {
+// 	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&m))
+// }
+
+// func (m MsgForceTransfer) GetSigners() []sdk.AccAddress {
+// 	sender, _ := sdk.AccAddressFromBech32(m.Sender)
+// 	return []sdk.AccAddress{sender}
+// }
```

Comment:
what is that?

Reply:
It comes from original module, so I do not know the reason why MsgForceTransfer was disabled in this version of tokenfactory.